static/app/views/detectors/hooks/useCreateDetectorFormSubmit.tsx

```diff
@@ -44,6 +46,18 @@ export function useCreateDetectorFormSubmit<
         const payload = formDataToEndpointPayload(data as TFormData);
         const resultDetector = await createDetector(payload);
 
+        const snubaQuery: SnubaQuery | undefined =
+          resultDetector.type === 'metric_issue'
+            ? resultDetector.dataSources[0]?.queryObj?.snubaQuery
+            : undefined;
+
+        trackAnalytics('monitor.created', {
+          organization,
+          detector_type: resultDetector.type,
+          dataset: snubaQuery?.dataset,
```

Comment:
Hmm this also might be annoying to deal with as we add more types. Maybe we can add some function on the config which would take a detector and return the properties that it should record for analytics?